Organising software projects task page (#226)
EVERSE-ResearchSoftware/RSQKit: pages/your_tasks/organising_software_project.md

---
title: How to organise your software project's directory
description: Good practice for organising files in a software project (or a research project in general) 
contributors: ["Aleksandra Nenadic"]
page_id: organising_software_project
related_pages:

>   your_tasks: [how_to_make_a_good_readme, licensing_software, releasing_code, software_documentation]

Reviewer: ```suggestion
  your_tasks: [how_to_make_a_good_readme, version_control, licensing_software, releasing_code, software_documentation]
```

---
title: How to organise your software project's directory
description: Good practice for organising files in a software project (or a research project in general) 
contributors: ["Aleksandra Nenadic"]
page_id: organising_software_project
related_pages:
  your_tasks: [how_to_make_a_good_readme, version_control, licensing_software, releasing_code, software_documentation]
---

> ## Why does it matter how we organise our software project's directory?

Author: ```suggestion
## Why does it matter how you organise your software project's directory?
```

Below is an example of a directory structure for a generic research project.

```
project_name/
├── README                # overview of the project
├── LICENSE               # license (reuse terms) for the project as a whole
├── data/                 # data files used in the project
│   ├── README            # describe the origin of your data
│   ├── raw/              # store your raw data and do not modify it
│   └── processed/        # store cleaned/processed/modified data separately 
├── manuscript/           # manuscript describing the results
├── results/              # results of the analysis (data, tables)  
│   ├── preliminary/
│   └── final/
├── figures/              # results of the analysis (figures)
│   ├── comparison_plot.png
│   └── regression_chart.pdf
├── src/                  # contains source code for the project
│   ├── LICENSE           # license that just applies to the code
│   ├── requirements.txt  # software requirements and dependencies
│   ├── main_script.py    # main script/code entry point
│   └── ...
├── doc/                  # documentation for your software
├── index.html            # entry point into the documentation website    
└── ...
```

> Checkout the {% tool "fair-python-coockiecutter" %} - a template tool that can help you set up a Python software project skeleton that uses modern state-of-the-art development tools and helps you follow best practices for code and metadata quality.

Author: ```suggestion
Checkout the {% tool 'fair-python-coockiecutter' %} - a template tool that can help you set up a Python software project skeleton that uses modern state-of-the-art development tools and helps you follow best practices for code and metadata quality.
```
